aws/resource_aws_appconfig_deployment.go

```diff
@@ -0,0 +1,216 @@
+)
+
+func resourceAwsAppconfigDeployment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppconfigDeploymentCreate,
+		Read:   resourceAwsAppconfigDeploymentRead,
+		Update: resourceAwsAppconfigDeploymentUpdate,
+		Delete: resourceAwsAppconfigDeploymentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"application_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"environment_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"deployment_strategy_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"configuration_profile_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"configuration_version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 1024),
+				),
+			},
+			"tags": tagsSchema(),
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"deployment_number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppconfigDeploymentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+	applicationID := aws.String(d.Get("application_id").(string))
+	environmentID := aws.String(d.Get("environment_id").(string))
+	deploymentStrategyID := aws.String(d.Get("deployment_strategy_id").(string))
+	configProfileID := aws.String(d.Get("configuration_profile_id").(string))
+	deploymentDescription := aws.String(d.Get("description").(string))
+	configVersion := aws.String(d.Get("configuration_version").(string))
+
+	input := &appconfig.StartDeploymentInput{
+		ApplicationId:          applicationID,
+		EnvironmentId:          environmentID,
+		DeploymentStrategyId:   deploymentStrategyID,
+		ConfigurationProfileId: configProfileID,
+		ConfigurationVersion:   configVersion,
+		Description:            deploymentDescription,
+		Tags:                   keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().AppconfigTags(),
+	}
+
+	var deploy *appconfig.StartDeploymentOutput
+	var err error
+	err = resource.Retry(2*time.Minute, func() *resource.RetryError {
+		deploy, err = conn.StartDeployment(input)
+		if err != nil {
+			if isAWSErr(err, appconfig.ErrCodeConflictException, "") {
+				log.Printf("[DEBUG] Retrying AppConfig Deployment creation: %s", err)
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		deploy, err = conn.StartDeployment(input)
+	}
+
+	if err != nil {
+		return fmt.Errorf("Creating AppConfig Deployment failed: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s-%s-%d", aws.StringValue(applicationID), aws.StringValue(environmentID), aws.Int64Value(deploy.DeploymentNumber)))
```

Comment:
how did you determine this O.O

Reply:
Unlike some of the other appconfig stuff, AWS doesn't provide an ID for deployments. I used applicationID, environmentID, and deployment number because thats the least amount of information you need to uniquely identify a deployment. I originally was going to go with just deployment number, but you could have deployment numbers repeated across different envs/applications.